core/src/banking_stage/committer.rs

```diff
@@ -104,12 +107,21 @@ impl Committer {
         let commit_transaction_statuses = tx_results
             .execution_results
             .iter()
-            .map(|execution_result| match execution_result.details() {
-                Some(details) => CommitTransactionDetails::Committed {
-                    compute_units: details.executed_units,
+            .zip(tx_results.loaded_accounts_stats.iter())
+            .map(
+                |(execution_result, loaded_accounts_stats)| match execution_result.details() {
+                    // reports actual execution CUs, and actual loaded accounts size for
+                    // transaction committed to block. qos_service uses these information to adjust
+                    // reserved block space.
+                    Some(details) => CommitTransactionDetails::Committed {
+                        compute_units: details.executed_units,
+                        loaded_accounts_data_size: loaded_accounts_stats
+                            .as_ref()
+                            .map_or(0, |stats| stats.loaded_accounts_data_size),
```

Comment:
When would we expect this to be None? Nop instructions/tx or would this also include committed but failed?

Reply:
Good question! `loaded_accounts_stats` is `vec<Result<>>` collected [here](https://github.com/anza-xyz/agave/blob/master/runtime/src/bank.rs#L4142-L4155), it'd carry error if failed to `load_accounts(tx...)`, in which case `0` is used as "not taken up any memory space". Otherwise the actual loaded accounts size is used.